packages/wonder-blocks-timing/components/action-scheduler-provider.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+import * as React from "react";
+import ActionScheduler from "../util/action-scheduler.js";
+
+import type {IScheduleActions} from "../util/types.js";
+
+type Props = {|
+    /**
+     * A function that, when given an instance of `IScheduleActions` will
+     * render a `React.Node`.
+     */
+    children: (IScheduleActions) => React.Node,
+|};
+
+/**
+ * A provider component that passes our action scheduling API to its children
+ * and ensures that all scheduled actions are cleared on unmount.
+ *
+ * ```jsx
+ * <ActionSchedulerProvider>
+ *     {actionScheduler => this.renderThingThatNeedsTimers(actionScheduler)}
```

Comment:
I'm seeing you are using `actionScheduler` here, and `schedule` in `withActionScheduler`. What do you think about using the same naming for both classes to keep consistency?

Reply:
Yeah, I'll use `schedule` here. The reason for the difference was that the HOC is the public-facing API and so I was being a little more aware of the time devs will take to type out what they need (`actionScheduler.timeout` is a lot worse than `setTimeout` - `schedule.timeout` seemed more reasonable).